plugins/window-rules/view-action-interface.cpp

```diff
@@ -338,6 +339,80 @@ std::tuple<bool, float> view_action_interface_t::_validate_alpha(
 std::tuple<bool, int, int, int, int> view_action_interface_t::_validate_geometry(
     const std::vector<variant_t> & args)
 {
+    if (args.size() == 2)
+    {
+        if (wf::is_string(args.at(1)))
+        {
+            // parse X11 style geometry widthxheight+x+y
+            wf::geometry_t workarea = _nontoplevel->get_output()->workarea->get_workarea();
+            wlr_box bb = _nontoplevel->get_bounding_box();
+            std::string geo = wf::get_string(args.at(1));
+            std::istringstream g(geo);
+            int x = bb.x, y = bb.y, w = bb.width, h = bb.height;
+            char c;
+
+            c = g.peek();
+            if ((c != 'x') && (c != '-') && (c != '+'))
+            {
+                g >> w;
+                if (g.fail())
+                {
+                    goto err;
```

Comment:
Not a fan of goto, I wouldn't add it in newer code. Also have you considered falling back to C-style sscanf for this? I think it would make parsing way simpler to read and follow.

Reply:
(this == the parsing of the geometry string)